src/kbmod/regionsearch/abstractions.py

```diff
@@ -0,0 +1,54 @@
+from abc import ABC, abstractmethod
+from typing import Type
+
+import numpy
+from astropy.coordinates import Angle, EarthLocation, SkyCoord
+from astropy.time import Time
+
+from kbmod.regionsearch.region_search import Filter
+
+
+class ObservationIndexer(ABC):
+    """
+    A mixin abstract class with a method to assign pointings to cluster indices.
+
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    @abstractmethod
+    def observations_to_indices(
+        self, pointing: SkyCoord, time: Time, fov: Angle, location: EarthLocation
+    ) -> numpy.ndarray:
+        """Returns a list of indices
```

Comment:
will do in next commit. The indices are groupings (clusters) of observations and the meaning is specific to the indexer. The PartitionIndexer divides the sky into exactly two clusters. Those that are within the search cone defined at instantiation are given the is_in_index and those that are not are given the is_out_index. Another indexer I have talked about is a HealpixIndexer which assigns observations to healpix regions identified by the hierarchical healpix id. Perhaps the way to think about indexing is for caches so that we do not have to go back to all the observation sources when searching a region. This is not obvious with this commit because the indexer is so specific.